src/models/static_models.jl

```diff
@@ -1,3 +1,6 @@
 abstract type StaticInjection <: Device end
-supports_services(d::Device) = true
+function supports_services(::T) where {T <: Device}
```

Comment:
I advise against this.  It imposes a field name requirement, which we've avoided doing.  The previous code mandated that every subtype of device implement a `get_services` method which means the field names could be anything.

Your solution is the simplest and perhaps the most pragmatic in the short term.  Here are some alternatives that may be better in the long run:
1. Leave the old `supports_services` method in place.  If any new device does not have services then it must implement a version that returns false.
2. Change the old `supports_services` method to call `error("not implemented")`.  Require that every subtype of Device be covered by an appropriate method.  This may be straightforward if there is a clean breakdown by abstract type.
3. Use your solution as a catch-all but implement `support_services` for types that we know support them and document that others should do the same.  That may not work out well in practice.